### ISSUE TITLE
There needs to be less dependence on Eidos as a whole

### ISSUE THREAD

**Issue by kwalcock** [clulab/ConceptAlignment#7]
What subset of Eidos do we need?  Presently we have the following imports:

```
  D:\Users\kwa\Documents\MyData\Projects\concept_alignment-project\ConceptAlignment\src\main\scala\org\clulab\alignment\groundings\DatamartNamer.scala (1 hit)
	Line 3: import org.clulab.wm.eidos.utils.Namer
  D:\Users\kwa\Documents\MyData\Projects\concept_alignment-project\ConceptAlignment\src\main\scala\org\clulab\alignment\groundings\GrounderApp.scala (2 hits)
	Line 7: import org.clulab.wm.eidos.groundings.EidosWordToVec
	Line 8: import org.clulab.wm.eidos.groundings.RealWordToVec
  D:\Users\kwa\Documents\MyData\Projects\concept_alignment-project\ConceptAlignment\src\main\scala\org\clulab\alignment\groundings\grounders\DatamartGrounder.scala (2 hits)
	Line 6: import org.clulab.wm.eidos.groundings.ConceptEmbedding
	Line 7: import org.clulab.wm.eidos.groundings.EidosWordToVec
  D:\Users\kwa\Documents\MyData\Projects\concept_alignment-project\ConceptAlignment\src\main\scala\org\clulab\alignment\groundings\Grounding.scala (1 hit)
	Line 3: import org.clulab.wm.eidos.utils.Namer
  D:\Users\kwa\Documents\MyData\Projects\concept_alignment-project\ConceptAlignment\src\main\scala\org\clulab\alignment\groundings\ontologies\DatamartOntology.scala (5 hits)
	Line 5: import org.clulab.wm.eidos.groundings.EidosWordToVec
	Line 6: import org.clulab.wm.eidos.utils.Closer.AutoCloser
	Line 7: import org.clulab.wm.eidos.utils.Namer
	Line 8: import org.clulab.wm.eidos.utils.Sourcer
	Line 9: import org.clulab.wm.eidos.utils.TsvReader
  D:\Users\kwa\Documents\MyData\Projects\concept_alignment-project\ConceptAlignment\src\main\scala\org\clulab\alignment\lucene\IndexerApp.scala (3 hits)
	Line 13: import org.clulab.wm.eidos.utils.Closer.AutoCloser
	Line 14: import org.clulab.wm.eidos.utils.Sourcer
	Line 15: import org.clulab.wm.eidos.utils.TsvReader
  D:\Users\kwa\Documents\MyData\Projects\concept_alignment-project\ConceptAlignment\src\main\scala\org\clulab\alignment\lucene\SearcherApp.scala (1 hit)
	Line 13: import org.clulab.wm.eidos.utils.Closer.AutoCloser
  D:\Users\kwa\Documents\MyData\Projects\concept_alignment-project\ConceptAlignment\src\main\scala\org\clulab\alignment\scraper\DatamartScraper.scala (1 hit)
	Line 4: import org.clulab.wm.eidos.utils.TsvWriter
  D:\Users\kwa\Documents\MyData\Projects\concept_alignment-project\ConceptAlignment\src\main\scala\org\clulab\alignment\scraper\IsiScraper.scala (1 hit)
	Line 7: import org.clulab.wm.eidos.utils.TsvWriter
  D:\Users\kwa\Documents\MyData\Projects\concept_alignment-project\ConceptAlignment\src\main\scala\org\clulab\alignment\scraper\NyuScraper.scala (1 hit)
	Line 4: import org.clulab.wm.eidos.utils.TsvWriter
  D:\Users\kwa\Documents\MyData\Projects\concept_alignment-project\ConceptAlignment\src\main\scala\org\clulab\alignment\scraper\ScraperApp.scala (3 hits)
	Line 4: import org.clulab.wm.eidos.utils.Closer.AutoCloser
	Line 5: import org.clulab.wm.eidos.utils.FileUtils
	Line 6: import org.clulab.wm.eidos.utils.TsvWriter
  D:\Users\kwa\Documents\MyData\Projects\concept_alignment-project\ConceptAlignment\src\main\scala\org\clulab\alignment\scraper\SuperMaasScraper.scala (1 hit)
	Line 4: import org.clulab.wm.eidos.utils.TsvWriter
  D:\Users\kwa\Documents\MyData\Projects\concept_alignment-project\ConceptAlignment\src\main\scala\org\clulab\alignment\utils\ConceptUtils.scala (1 hit)
	Line 5: import org.clulab.wm.eidos.groundings.{EidosOntologyGrounder, OntologyHandler}
  D:\Users\kwa\Documents\MyData\Projects\concept_alignment-project\ConceptAlignment\src\main\scala\org\clulab\alignment\utils\PropertiesBuilder.scala (2 hits)
	Line 5: import org.clulab.wm.eidos.utils.Closer.AutoCloser
	Line 6: import org.clulab.wm.eidos.utils.FileUtils
```

**Comment by kwalcock** [comment]
The critical ones seem to be
```
	Line 7: import org.clulab.wm.eidos.groundings.EidosWordToVec
	Line 8: import org.clulab.wm.eidos.groundings.RealWordToVec
	Line 5: import org.clulab.wm.eidos.groundings.{EidosOntologyGrounder, OntologyHandler}
```
while all the others are utils whose code could easily be duplicated in order to save us the dependency.

`OntologyHandler` implies `EidosOntologyGrounder` implies `EidosWordToVec`.  The concrete implementation of `EidosWordToVec` is `RealWordToVec`.
  

ConceptEmbedding is one line:
```
case class ConceptEmbedding(val namer: Namer, embedding: Array[Float]) extends Serializable
```

Namer is this:
```
trait Namer {
  def name: String // gets the entire name with parts separated by /
  def branch: Option[String] // gets the branch in top/branch/[more/]leaf

  override def toString: String = name
}
```

**Comment by kwalcock** [comment]
I did some rearranging so that the only dependency is
```
  D:\Users\kwa\Documents\MyData\Projects\concept_alignment-project\ConceptAlignment\src\main\scala\org\clulab\alignment\utils\ConceptUtils.scala (1 hit)
	Line 5: import org.clulab.wm.eidos.groundings.{EidosOntologyGrounder, OntologyHandler}
```

**Comment by kwalcock** [comment]
The largest component of Eidos is glove.  We're using glove anyway.  It doesn't seem like splitting up Eidos would help much in this case.

**Comment by kwalcock** [comment]
We might not need to use glove, at least not in the webapp.  I propose that any top-down ontology that the webapp has access to be pre-indexed, just as the bottom-up ontologies have been pre-indexed.  The pre-indexing and only that would depend on Eidos.  The main ConceptOntology project is independent of Eidos.  An indexer subproject depends on Eidos and main.  The webapp depends on only main and the output of the indexers.  The indexer for an ontology converts what is usually a hierarchical yaml ontology into a tsv table (or similar) with columns for the node name, perhaps the branch, and then the vectors, very much like the indexer for the bottom up ontologies.  In fact, it could go through annoy as well.  All the webapp would do is provide a lucene way to search for texts and then the KNN algorithm.  It will know next to nothing about ontologies and glove.

**Comment by MihaiSurdeanu** [comment]
+1

**Comment by kwalcock** [comment]
I'm making subprojects then.  @zhengzhongliang and @BeckySharp might not want to make lots of changes at the same time.

**Comment by BeckySharp** [comment]
given how slow i am, shouldn't be a problem!
should i still remove the app that needs eidos?

**Comment by kwalcock** [comment]
No and no.  BTW I wondered if an issue would be adequate for discussing the terminology.

Now that we have Entity and Concept in some of the code, next up is a Thing class with instances thing1 and thing2.

**Comment by BeckySharp** [comment]
hahaha YES!

**Comment by kwalcock** [comment]
I think this issue has been addressed.  Zhengzhong's code was recently merged into its own subproject with #11 and #12.
